src/entities/activity/ui/items/CheckboxItem/PortraitCheckboxOption.tsx

```diff
@@ -0,0 +1,96 @@
+import { useMemo } from 'react';
+
+import { Box } from '~/shared/ui';
+import {
+  CheckboxItem,
+  CustomTooltip,
+  SelectBaseBox,
+  SelectBaseImage,
+  SelectBaseText,
+} from '~/shared/ui';
+import { cutString } from '~/shared/utils';
+
+type Props = {
+  id: string;
+  name: string;
+  value: string | number;
+  label: string;
+
+  description: string | null;
+  image: string | null;
+  disabled?: boolean;
+  defaultChecked?: boolean;
+  color: string | null;
+
+  onChange: () => void;
+  replaceText: (value: string) => string;
+};
+
+export const PortraitCheckboxOption = (props: Props) => {
+  const tooltipText = useMemo(() => {
+    if (props.description) {
+      return props.replaceText(props.description);
+    }
+
+    return null;
+  }, [props]);
+
+  const labelText = useMemo(() => {
+    return props.replaceText(props.label);
+  }, [props]);
+
+  const cutTheTextBasedOnOptionParams = (text: string) => {
+    if (props.image && !tooltipText) {
+      return cutString(text, 21);
+    }
+
+    if (props.image && tooltipText) {
+      return cutString(text, 12);
+    }
+
+    return cutString(text, 25);
+  };
```

Comment:
Can you use [line-clamp](https://developer.mozilla.org/en-US/docs/Web/CSS/-webkit-line-clamp) though?

Reply:
I am not sure about it. Seems like this feature has very bad availability.

<img width="678" alt="Screenshot 2024-07-24 at 13 26 46" src="https://github.com/user-attachments/assets/17a2bdab-69db-4e95-b7b8-5fa9e4627bcb">

Reply:
However, if you click "See full compatibility" it will lead you to the screen:
<img width="799" alt="image" src="https://github.com/user-attachments/assets/35a16838-5cf4-43bf-bedc-96944ad3f1cc">
It is saying that "none" value for line-clamp is weakly supported indeed. However, as for line-clamp itself this is not the case. It's supported across all popular browsers and only has a note to use it with "-webkit-" prefix: -webkit-line-clamp.

Another good source to check a feature support across browsers is https://caniuse.com/.
For instance:
<img width="1420" alt="image" src="https://github.com/user-attachments/assets/06e88a10-8920-42ae-b83c-9a3f49029fd3">